src/sidebar/components/PendingUpdatesNotification.tsx

```diff
@@ -59,32 +81,32 @@ function PendingUpdatesNotification({
     <div
       role="status"
       className="absolute right-0 animate-updates-notification-slide-in"
+      data-testid="notification-container"
     >
       <Button
         onClick={applyPendingUpdates}
         unstyled
         classes={classnames(
           'flex gap-1.5 items-center py-1 px-2',
           'rounded shadow-lg bg-gray-900 text-white',
+          'group focus-visible-ring',
         )}
-        onMouseEnter={() => setCollapsed(false)}
-        onFocus={() => setCollapsed(false)}
-        onMouseLeave={() => !timeout.current && setCollapsed(true)}
-        onBlur={() => !timeout.current && setCollapsed(true)}
+        data-testid="notification-button"
       >
         <DownloadIcon className="w-em h-em opacity-80" />
-        {!collapsed && (
-          <span data-testid="full-notification" className="whitespace-nowrap">
-            Load <span className="font-bold">{pendingUpdateCount}</span>{' '}
-            {pluralize(pendingUpdateCount, 'update', 'updates')}{' '}
-            <span className="sr-only">by pressing l</span>
-          </span>
-        )}
-        {collapsed && (
-          <span data-testid="collapsed-notification" className="font-bold">
-            {pendingUpdateCount}
-          </span>
-        )}
+        <div
+          data-testid={
+            collapsed ? 'collapsed-notification' : 'full-notification'
+          }
+          className="flex gap-1"
+        >
+          <HideableBlock hidden={collapsed}>Load</HideableBlock>
+          <span className="font-bold">{pendingUpdateCount}</span>
+          <HideableBlock hidden={collapsed}>
+            {pluralize(pendingUpdateCount, 'update', 'updates')}
+          </HideableBlock>
+        </div>
+        <span className="sr-only">by pressing l</span>
```

Comment:
By no longer conditionally rendering the collapsed or expanded blocks and using conditional CSS classes instead, we should reduce potential mouse leave events triggered in the button when hovering over a children element.

Reply:
The MDN [documentation for `mouseleave`](https://developer.mozilla.org/en-US/docs/Web/API/Element/mouseleave_event#specifications) (which I checked matches the spec) states:

> mouseleave and [mouseout](https://developer.mozilla.org/en-US/docs/Web/API/Element/mouseout_event) are similar but differ in that mouseleave does not bubble and mouseout does. This means that mouseleave is fired when the pointer has exited the element and all of its descendants, whereas mouseout is fired when the pointer leaves the element or leaves one of the element's descendants (even if the pointer is still within the element).

In other words, `mouseleave` should never fire when a mouse exits a descendant. Only when it exits the outer element.

Did you observe `mouseleave` behaving differently?

Reply:
My assumptions are based on some references I found about `onMouseLeave` not being reliable in React https://github.com/facebook/react/issues/4492

However, we don't use react, so maybe this does not apply to us.

As per reproducing it, I was able to do it only one, and Don told be it happened to him as well, but couldn't show me when screen-sharing.

I don't know what were the exact conditions in my case, but I found that, when moving the mouse over the collapsed notification, it made it expand, but if I moved the mouse towards the number of pending notifications, it collapsed again.

I was able to reproduce it only once when I was working on something else.

One potential reason could be that a new update happened while I was hovering the notification, and I don't know if that could have caused the sync between references to elements in the DOM vs VDOM to be lost somehow. But this is pure speculation, as I don't know how that works internally.

In the end, I decided to leave this open since, as you mentioned, these changes are reasonable anyway, even if we can't fully reproduce the issue in order to verify it.

Reply:
The React issue you linked to makes reference to `mouseout` events, which made it sound as if it wasn't using the native `mouseleave` event, and sure enough https://github.com/facebook/react/issues/11972#issuecomment-876790218 backs this up.

> I don't know what were the exact conditions in my case, but I found that, when moving the mouse over the collapsed notification, it expanded as expected, but if I then moved the mouse towards the number of pending notifications, it collapsed again.

OK. Well looking at the code on main, I can see how this happens without any misbehavior from `mouseleave` events:

1. A pending update occurs, and starts a timeout
2. User hovers the notification before the timeout expires
3. The timeout expires, and the callback invokes `setCollapsed(true)`

The mistake here is that we have only one state variable tracking two separate things, from which the collapsed state should be computed:
1. Whether the component is hovered
2. Whether the initial timer is active

The change to CSS on this branch fixes the problem, because it eliminates the need for (1).